cpp/test/sg/fil_test.cu

```diff
@@ -754,28 +756,29 @@ class TreeliteFilTest : public BaseFilTest<float> {
   {
     int key = (*pkey)++;
     builder->CreateNode(key);
-    const fil::dense_node<float>& dense_node = nodes[node];
+    const fil::dense_node<real_t>& dense_node = this->nodes[node];
     std::vector<std::uint32_t> left_categories;
     if (dense_node.is_leaf()) {
-      switch (ps.leaf_algo) {
+      switch (this->ps.leaf_algo) {
         case fil::leaf_algo_t::FLOAT_UNARY_BINARY:
         case fil::leaf_algo_t::GROVE_PER_CLASS:
           // default is fil::FLOAT_UNARY_BINARY
-          builder->SetLeafNode(key, tlf::Value::Create(dense_node.output<float>()));
+          builder->SetLeafNode(key, tlf::Value::Create(dense_node.template output<real_t>()));
           break;
         case fil::leaf_algo_t::CATEGORICAL_LEAF: {
-          std::vector<tlf::Value> vec(ps.num_classes);
-          for (int i = 0; i < ps.num_classes; ++i) {
-            vec[i] = tlf::Value::Create(i == dense_node.output<int>() ? 1.0f : 0.0f);
+          std::vector<tlf::Value> vec(this->ps.num_classes);
+          for (int i = 0; i < this->ps.num_classes; ++i) {
+            vec[i] =
+              tlf::Value::Create(i == dense_node.template output<int>() ? real_t(1) : real_t(0));
           }
           builder->SetLeafVectorNode(key, vec);
           break;
         }
         case fil::leaf_algo_t::VECTOR_LEAF: {
-          std::vector<tlf::Value> vec(ps.num_classes);
-          for (int i = 0; i < ps.num_classes; ++i) {
-            auto idx = dense_node.output<int>();
-            vec[i]   = tlf::Value::Create(vector_leaf[idx * ps.num_classes + i]);
+          std::vector<tlf::Value> vec(this->ps.num_classes);
+          for (int i = 0; i < this->ps.num_classes; ++i) {
+            auto idx = dense_node.template output<int>();
+            vec[i]   = tlf::Value::Create(this->vector_leaf[idx * this->ps.num_classes + i]);
```

Comment:
Same question for these `this` references and any others which have been introduced.